cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/Result.kt

```diff
@@ -0,0 +1,274 @@
+package com.thefuntasty.mvvm.crusecases
+
+import kotlinx.coroutines.CancellationException
+
+/**
+ * A discriminated union that encapsulates successful outcome either with [Success] or with [Error]
+ *
+ * This class is adapted to kotlin coroutines so if [Error] contains [CancellationException] then this error is rethrown.
+ */
+sealed class Result<out VALUE : Any?> {
+    override fun toString(): String {
+        return when (this) {
+            is Success -> "Success($value)"
+            is Error -> "Error(${error})"
+        }
+    }
+
+    operator fun component1(): VALUE? {
+        return if (this is Success) value else null
+    }
+
+    operator fun component2(): Throwable? {
```

Comment:
<table data-meta="generated_by_danger-pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/1e22aaaf9def5bbeda391f2cd9735be2fc018fbd/cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/Result.kt#L22"></span>Functions with exact one statement, the return statement, can be rewritten with ExpressionBodySyntax.</td>
    </tr>
  </tbody>
</table>